ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/LedgerTimeHelper.scala

```diff
@@ -75,6 +75,12 @@ final case class LedgerTimeHelper(
                 else
                   Future.successful(Left(ErrorCause.LedgerTime(maxRetries)))
               })
+              .recoverWith {
+                // An error while looking up the maximum ledger time for the used contracts
+                // most likely means that one of the contracts is already not active anymore,
+                // which can happen under contention. Retry without advancing time in this case.
+                case _ => loop(commands, submissionSeed, retriesLeft - 1)
```

Comment:
@rautenrieth-da: the catch-all case seem problematic. It will hide errors in the future. For safety: could at leats log at INFO level the error that happened and that you are retrying? Also consider doing a case distinctions on the errors and going through them one-by-one to decide whether the chosen action is the appropriate one.

Reply:
Another point to consider: could it be that a retry is "in too quick succession", i.e., that some outside update to the index-store needs to happen for the retry to work? In that case, we might want to consider an exponential back-off strategy to avoid busy-waiting.

Reply:
This error should only happen if the ACS changed between command interpretation and the lookup of maximum ledger time of used contracts, which happens _immediately_ after command interpretation.

An immediate retry would therefore already use a different ACS than the original interpretation. However, as @gerolf-da noted, if the transaction uses fixed contract IDs (as opposed those looked by by key), then a retry would not change the result.

If fixed contract IDs are used, and some of those contracts are archived between command interpretation and the subsequent ledger time lookup:
1. The first submission would return `ABORTED`, signalling that no ledger time could be found, and suggesting the app to retry.
2. The second submission would return the final error `INVALID_ARGUMENT`, because the archived contracts do not exist at command interpretation.

IMO that is a reasonable behavior.